Error checking when adding new config var

mnlevy1981 opened the pull request:
When adding a new configuration variable or parameter, abort if the short name
(used for the put and get statements) is not unique or if the pointer provided
points to memory already registered with a different config var or parameter.

mnlevy1981 committed:
Error checking when adding new config var

When adding a new configuration variable or parameter, abort if the short name
(used for the put and get statements) is not unique or if the pointer provided
points to memory already registered with a different config var or parameter.

mnlevy1981 committed:
Forgot to actually update marbl_status_log

When a configuration variable or parameter was pointing at same variable in
memory as an existing variable, last commit would write an appropriate log
message but never actually add that message to the status log.

mnlevy1981 commented:
Samples of the two error messages...

Duplicating a short name when defining a parameter (triggered by commenting out the updated short name):

```
MARBL ERROR (marbl_config_mod:marbl_var_add): parm_Fe_bioavail has been added twice
MARBL ERROR (marbl_parms:marbl_define_parameters): Error reported from this%add_var(parm_Fe_bioavail)
MARBL ERROR (marbl_interface:init): Error reported from marbl_define_parameters()
MARBL ERROR (marbl_init_namelist_drv:test): Error reported from marbl%init
```

Duplicating a pointer when defining a parameter (triggered by commenting out the updated pointer):

```
MARBL ERROR (marbl_config_mod:marbl_var_add): parm_o2_min and parm_Fe_bioavail both point to same variable in memory.
MARBL ERROR (marbl_parms:marbl_define_parameters): Error reported from this%add_var(parm_o2_min)
MARBL ERROR (marbl_interface:init): Error reported from marbl_define_parameters()
MARBL ERROR (marbl_init_namelist_drv:test): Error reported from marbl%init
```

And both errors can be reported simultanously (the abort isn't triggered until after the error checking):

```
MARBL ERROR (marbl_config_mod:marbl_var_add): parm_Fe_bioavail has been added twice
MARBL ERROR (marbl_config_mod:marbl_var_add): parm_Fe_bioavail and parm_Fe_bioavail both point to same variable in memory.
MARBL ERROR (marbl_parms:marbl_define_parameters): Error reported from this%add_var(parm_Fe_bioavail)
MARBL ERROR (marbl_interface:init): Error reported from marbl_define_parameters()
MARBL ERROR (marbl_init_namelist_drv:test): Error reported from marbl%init
```

klindsay28 reviewed (approved):
Please correct numbering in blocks of marbl_var_add.

Please add comment in blocks 3 & 4 that pointers are nullified in the class definition of new_vars.

mnlevy1981 committed:
Improve comments per code review

Enumerated list in comments repeated item 3 (1, 2, 3, 3, 4 instead of 1,
2, 3, 4, 5) and we thought it would be useful to explain how the
logical pointers in a derived type were nullified in the type
definition.
